### PR TITLE
Merge master to release 3.0.0

### PR DESCRIPTION
**Pull request by inodb**

### CODE REVIEW

File: web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java [hunk 1]
```diff
@@ -137,6 +137,64 @@
         return true;
     }
 
+    private boolean extractAttributesFromClinicalDataBinCountFilter(HttpServletRequest request) {
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java]: Isn't this function already defined on line 443? This seems like a second copy

File: web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java [hunk 2]
```diff
@@ -137,6 +137,64 @@
...
+                request.setAttribute("involvedCancerStudies", cancerStudyIdCollection);
+            }
+        } catch (Exception e) {
+            LOG.error("exception thrown during extraction of clinicalDataBinCountFilter: " + e);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean extractAttributesFromClinicalDataCountFilter(HttpServletRequest request) {
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java]: Isn't this function already defined on 463? This is a second copy

File: web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java [hunk 3]
```diff
@@ -137,6 +137,64 @@
...
+                request.setAttribute("involvedCancerStudies", cancerStudyIdCollection);
+            }
+        } catch (Exception e) {
+            LOG.error("exception thrown during extraction of clinicalDataBinCountFilter: " + e);
+            return false;
+        }
+        return true;
+    }
+
+    private boolean extractAttributesFromStudyViewFilter(HttpServletRequest request) {
```
Comment (sheridancbio) [inline web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java]: Another copy is on line 504 .. I do notice a bug in both of these ... the Log.error() reports clinicalDataBinCountFilter, but this (and the previous function) are parsing different types. But a bugfix should be in a separate PR.

**Comment by inodb** [comment]
@sheridancbio 	this is fixed now

File: web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java (hunk 3 above)
Comment (inodb) [reply web/src/main/java/org/cbioportal/web/util/InvolvedCancerStudyExtractorInterceptor.java]: didn't fix this - we can address it in a separate PR as suggested

**Comment by sheridancbio** [comment]
Looks good @inodb 👍
